framework/src/bcs/NeumannBC.C

```diff
@@ -18,13 +18,13 @@ template<>
 InputParameters validParams<NeumannBC>()
 {
   InputParameters params = validParams<IntegratedBC>();
-  params.addParam<Real>("value", 0.0, "The value of the gradient on the boundary.");
+  params.addParam<PostprocessorName>("value", 0.0, "The value of the gradient on the boundary.");
```

Comment:
I don't like this for one of our most basic classes in MOOSE.  This class is used (and inherited from) by EVERYONE... and now it's going to show that the `value` is expecting a `PostprocessorName`?  And now Peacock is going to suggest the value be an Postprocessor in your input file?

The right thing to do here is to make a custom BC that does this in your example.